src/passes/Flatten.cpp

```diff
@@ -347,6 +382,10 @@ struct Flatten
                 << type;
       }
     }
+
+    // Flatten can generate blocks within 'catch', making pops invalid. Fix them
+    // up.
+    EHUtils::handleBlockNestedPops(curr, *getModule());
```

Comment:
It looks like `handleBlockNestedPops` will scan for Trys without checking the features first. How about adding an early exit there if exceptions are not enabled?

Reply:
Done at the beginning of `handleBlockNestedPops`.